### ISSUE TITLE
NanoMQ Session Takeover Mechanism

### ISSUE THREAD

**Issue by Sea-Intestine-Rice** [nanomq/nanomq#1927]
**Describe the bug**
When two clients attempt to connect to the NanoMQ broker using the same ClientID, the broker should send a DISCONNECT packet with the Reason Code 0x8E (Session taken over) to the first connection. However,  I observed that NanoMQ did not send a DISCONNECT packet to the initial connection as expected.

**Expected behavior**
The broker should detect the session takeover when a second connection with the same ClientID is established and should send a DISCONNECT packet with the Reason Code 0x8E to the first client.

**Actual Behavior**
The broker did not send the DISCONNECT packet to the first client (sock1) when a second client (sock2) connected with the same ClientID. Instead, both clients remained connected without any indication of session takeover for the first connection.

**To Reproduce**
 1. Start the NanoMQ broker.
 2. Establish the first connection (sock1) to the broker with a connect packet using a specific ClientID.
   hexstream for sock1: `101900044d5154540502003c03210014000974616b652d6f766572`
 3. Establish the second connection (sock2) to the broker with the same connect packet, using the same ClientID.
 4. Observe the broker's response. The expected behavior is that the broker should send a DISCONNECT packet to sock1 with the Reason Code 0x8E.
 
**Environment Details**

- NanoMQ version 0.22.8
- Operating system and version  Ubuntu 22.04 LTS
- Compiler and language used: C/C++
- Testing scenario  Testing session takeover functionality by connecting two clients with the same ClientID and observing the broker's response.

**Client SDK**
Test cases were written using python.

**Additional context**
[take_over.zip](https://github.com/user-attachments/files/17532790/take_over.zip)

**Comment by JaylinYu** [comment]
Nope, the actual behaivour of nanomq is to kick the preivous client. Not sure the exact way of your testing, at least it works normally at other user's case.

But it indeed not send a Disconnect packet when session take over happens. But this is not compulsory in MQTT protocol.
 I have labeled it as an enhancement issue.

**Comment by Sea-Intestine-Rice** [comment]
Thank you for reviewing the issue. I’d like to reference the MQTT specification, which states: “If the ClientID represents a Client already connected to the Server, the Server sends a DISCONNECT packet to the existing Client with Reason Code of 0x8E (Session taken over) as described in section 4.13 and MUST close the Network Connection of the existing Client.”

This suggests that sending a DISCONNECT packet with Reason Code 0x8E to the first client during a session takeover is a required part of the protocol, ensuring compliance and clarity in client connection states. Our pcap file captures the complete interaction, showing the TCP session closing without this DISCONNECT packet.

Hope this behavior could be considered as an enhancement for better session management.

**Comment by JaylinYu** [comment]
Yep， I mean nanomq does close the previous connection that has a conflict clientid. You say it remains at issue repor.

Will plan this enhancment in future ver
